cmd/analyze/main.go

```diff
@@ -176,50 +177,51 @@ func main() {
 	flag.Parse()
 
 	if err := featureflags.Update(*features); err != nil {
-		slog.Error("Failed to parse flags", "error", err)
-		return
+		return -1, fmt.Errorf("failed to parse flags: %w", err)
```

Comment:
I think you want to stick to exit values of between 0-255. Although an `int` is acceptable only the lower 8 bits of the value are returned by some functions.

The docs for [`os.Exit`](https://pkg.go.dev/os#Exit) specifically suggest limiting the range to 0-125.

There is some useful detail here: https://unix.stackexchange.com/questions/418784/what-is-the-min-and-max-values-of-exit-codes-in-linux

For example `log.Fatal` (see [src](https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/log/log.go;l=413)) calls `os.Exit(1)`.

Reply:
Sure, I will change it to 2 as per Andrew's comment